build/ci.go

```diff
@@ -345,6 +347,67 @@ func downloadSpecTestFixtures(csdb *build.ChecksumDB, cachedir string) string {
 	return filepath.Join(cachedir, base)
 }
 
+func hashFile(path string) (common.Hash, error) {
+	f, err := os.OpenFile(path, os.O_RDONLY, 0666)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	contents, err := io.ReadAll(f)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return common.BytesToHash(crypto.Keccak256(contents)), nil
+}
+
+func generateCheck() {
+	tc := new(build.GoToolchain)
+	preHashes := make(map[string]common.Hash)
+	err := filepath.WalkDir(".", func(path string, d os.DirEntry, err error) error {
+		if strings.HasPrefix(path, "tests/testdata") {
```

Comment:
Note that path is a filepath, i.e. it has a platform-specific delimiter. In order to work correctly on windows, you need to convert your skip prefix into a platform path using `filepath.FromSlash`.

Also note you can return `filepath.SkipDir` from this function to ensure `WalkDir` does not visit the subdirectories of the skipped one.